### ISSUE TITLE
Possible to get updater.sh to work with symlinks?

### ISSUE THREAD

**Issue by adrian5** [arkenfox/user.js#1150]
I'd like to track my `user-overrides.js` in my dotfiles and symlink that into the Firefox profile.

Unfortunately, reading through symlinks doesn't appear to be supported.

---

Ideally, it would be nice if the `updater.sh` script itself could be run through a symlink too, so that one could quickly set up a patched `user.js` without browsing the web for this script first.

**Comment by earthlng** [comment]
Hi,

AFAIK `test -f` matches a regular file as well as a symlink to a file and therefore a symlinked user-overrides.js should already work. Can you provide more details as to how exactly you tried to run it and why it apparently didn't work?

The updater.sh itself also supports symlinking but the other way around where the symlink target needs to be the updater.sh in the profile directory. The idea here is that users can place the updater in their profiles and create a symlink on the Desktop to more easily run it, similar to how people on Windows like to put shortcuts on the Desktop for quick access.

I don't plan to implement better symlink support but pull requests are always welcome if someone else wants to do that.

> so that one could quickly set up a patched user.js without browsing the web for this script first.

have you looked at the available command line parameters?
You can already easily do that with the `-l` and `-o` parameters and without needing to copy or symlink anything. fe:

create a new profile with `firefox -P` and then run `/tmp/updater.sh -l -o /path/to/any/user-overrides.js` where the updater and overrides can be located anywhere you want.
`-l` will parse the profiles.ini and let you select a profile to update and then, once selected, the script will cd into that profile folder, download the latest user.js and apply the overrides file from the location you specify with `-o`.

**Comment by adrian5** [comment]
Sorry, I totally overlooked those flags! That gives me more than enough leeway to work without symlinks.

> Can you provide more details as to how exactly you tried to run it and why it apparently didn't work?

I ran the `updater.sh` itself through a symlink from within the Firefox profile directory, and the script then behaved as if the symlink-target (in my dotfiles) was the working directory. I think that's where the malfunction came from.

Instead of specifying files to link or copy, I now added a simple install command to my dotfile manager:

```
~/.dotfiles/arkenfox/updater.sh -blsu -o ~/.dotfiles/arkenfox/user-overrides.js
```

This has the added benefit that I don't need to write my own logic for finding the Firefox profile. The script prompts me and does all the necessary updating and merging; couldn't ask for more!

Thanks a lot, I'll close this issue.
